### ISSUE TITLE
PDFtk as optional/peer dependency?

### ISSUE THREAD

**Issue by moravcik** [corymickelson/CommonPdf#14]
Hello, thank you for your lib, we are actually using it for signing PDFs. 

As I can see there is a large dependency tree introduced by `pdftk`, but nothing of it is actually used for just signing PDF. This could be fixed by marking `pdftk` and `commonpdf_pdftk` as optional/peer dependency or just create dedicated module for signing PDFs.

The same should apply for `commonpdf_podofo` as well, if someone is not interested in signing PDFs.

What do you think?

**Comment by corymickelson** [comment]
@moravcik  Yes this has been something I've had in mind for a while, exposing some option and build script for "compiling" CommonPdf for specific use cases. I honestly have no idea when I will be able to get to this, it could take a couple months.
